plugin/manager_linux.go

```diff
@@ -199,9 +199,25 @@ func (pm *Manager) upgradePlugin(p *v2.Plugin, configDigest digest.Digest, blobs
 
 	pdir := filepath.Join(pm.config.Root, p.PluginObj.ID)
 	orig := filepath.Join(pdir, "rootfs")
+
+	// Make sure nothing is mounted
+	// This could happen if the plugin was disabled with `-f` with active mounts.
+	// If the anything in `orig` is still mounted, this should error out.
+	mounts, err := getMounts(orig)
+	if err != nil {
+		return err
+	}
+	for _, m := range mounts {
+		if err := mount.Unmount(m); err != nil {
+			if mounted, e := mount.Mounted(m); e != nil || mounted {
+				return errors.Wrap(err, "error unmounting plugin root during upgrade")
```

Comment:
I didn't want to do that because we really can't proceed if there is anything mounted.
This is what we do (sort of) when disabling the plugin, but for upgrade we really need to error out.